cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java

```diff
@@ -253,10 +254,10 @@ public void setTTL(String streamId, long ttlInSeconds)
    * @return list of {@link StreamRecord}s
    * @throws IOException if a network error occurred
    */
-  public List<StreamRecord> list() throws IOException, UnauthorizedException {
+  public List<StreamSpecification> list() throws IOException, UnauthorizedException {
     URL url = config.resolveNamespacedURLV3("streams");
     HttpResponse response = restClient.execute(HttpMethod.GET, url, config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<StreamRecord>>() { }).getResponseObject();
+    return ObjectResponse.fromJsonBody(response, new TypeToken<List<StreamSpecification>>() { }).getResponseObject();
```

Comment:
actually it is a StreamDetail... seems we have some duplication in our classes. See AppFabricDataHttpHandler